services/src/main/java/org/fao/geonet/api/processing/MInspireEtfValidateProcess.java

```diff
@@ -0,0 +1,209 @@
+        AccessManager accessManager = appContext.getBean(AccessManager.class);
+        InspireValidatorUtils inspireValidatorUtils = appContext.getBean(InspireValidatorUtils.class);
+        SchemaManager schemaManager = appContext.getBean(SchemaManager.class);
+        DataManager dataManager = appContext.getBean(DataManager.class);
+
+        metadataToAnalyseCount = uuids.size();
+        analyseMdDate = System.currentTimeMillis();
+
+        for (String uuid : uuids) {
+            if (!metadataRepository.existsMetadataUuid(uuid)) {
+                metadataAnalysed++;
+                metadataNotAllowed++;
+                continue;
+            }
+
+            for (AbstractMetadata record : metadataRepository.findAllByUuid(uuid)) {
+                try {
+                    if (!accessManager.canEdit(serviceContext, String.valueOf(record.getId()))) {
+                        metadataNotAllowed++;
+                    } else {
+                        runInNewTransaction("minspireetfvalidate-process-metadata", new TransactionTask<Object>() {
+                            @Override
+                            public Object doInTransaction(TransactionStatus transaction) throws Throwable {
+                                // Evaluate test conditions for INSPIRE test suites, to apply to the metadata
+                                Map<String, String> testsuiteConditions = inspireValidatorUtils.getTestsuitesConditions();
+
+                                try {
+                                    boolean inspireMetadata = false;
+
+                                    for (Map.Entry<String, String> entry : testsuiteConditions.entrySet()) {
+                                        boolean applyCondition = Xml.selectBoolean(record.getXmlData(false),
+                                            entry.getValue(),
+                                            schemaManager.getSchema(record.getDataInfo().getSchemaId()).getNamespaces());
+                                        if (applyCondition) {
+                                            String testId = inspireValidatorUtils.submitFile(serviceContext, URL,
+                                                new ByteArrayInputStream(record.getData().getBytes()), entry.getKey(), record.getUuid());
+
+                                            inspireValidatorUtils.waitUntilReady(serviceContext, URL, testId);
+
+                                            String reportUrl = inspireValidatorUtils.getReportUrl(URL, testId);
+                                            String reportXmlUrl = inspireValidatorUtils.getReportUrlXML(URL, testId);
+                                            String reportXml = inspireValidatorUtils.retrieveReport(serviceContext, reportXmlUrl);
+
+                                            boolean isValid = inspireValidatorUtils.isPassed(serviceContext, URL, testId)
```

Comment:
Maybe undefined should be considered not validate ?

![image](https://user-images.githubusercontent.com/1701393/71111408-a8a8cf80-21c9-11ea-9db0-5702ba76ef55.png)